test/functional/page_objects/visualize_page.js

```diff
@@ -594,11 +594,35 @@ export function VisualizePageProvider({ getService, getPageObjects }) {
     }
 
     async clickGo() {
+      const prevRenderingCount = await this.getVisualizationRenderingCount();
+      log.debug(`Before Rendering count ${prevRenderingCount}`);
       await testSubjects.clickWhenNotDisabled('visualizeEditorRenderButton');
-      await PageObjects.header.waitUntilLoadingHasFinished();
-      // For some reason there are two `data-render-complete` tags on each visualization in the visualize page.
-      const countOfDataCompleteFlags = 2;
-      await renderable.waitForRender(countOfDataCompleteFlags);
+      await this.waitForRenderingCount(prevRenderingCount);
+    }
+
+    async getVisualizationRenderingCount() {
+      const visualizationLoader = await testSubjects.find('visualizationLoader');
+      const renderingCount = await visualizationLoader.getAttribute('data-rendering-count');
+      return Number(renderingCount);
+    }
+
+    async waitForRenderingCount(previousCount = 0, increment = 1) {
+      await retry.try(async () => {
+        const currentRenderingCount = await this.getVisualizationRenderingCount();
+        log.debug(`Readed rendering count ${previousCount} ${currentRenderingCount}`);
+        expect(currentRenderingCount).to.be(previousCount + increment);
+      });
+    }
+
+    async waitForVisualizationRenderingCompleted() {
+      //assuming rendering is done when data-rendering-count is constant within 500 ms
+      await retry.try(async () => {
+        const previousCount = await this.getVisualizationRenderingCount();
+        await PageObjects.common.sleep(500);
```

Comment:
My only concern is that a rendering could take longer than 500ms and this would give a false complete indication.  I don't think it would hurt to wait a whole second.  But if you have enough data from test runs to indicate this covers it, I'm OK with leaving it.  (could always increase it later if needed).